### PR TITLE
[tests] Add pre-existing tests to managed test suite

### PR DESCRIPTION
**Pull request by tritao**

### CODE REVIEW

**Comment by tritao** [comment]
The idea here is to share the managed.dll test project between both generators. I'll be adding some more commits to the PR to port the C and Java test runner to `managed.dll`.

**Comment by spouliot** [comment]
@tritao I'm fine with that goal :) but the `objc` branch must remain green and unused code/tests needs to be identified as such.

**Comment by tritao** [comment]
Makes sense of course, I'll fix it the way you suggested.

**Comment by tritao** [comment]
Updated PR with feedback.

File: tests/managed/types.cs [hunk 1]
```diff
@@ -80,3 +80,40 @@ public static class Type_String
...
+	public sbyte PassAndReturnsSByte (sbyte v) { return v; }
+	public byte PassAndReturnsByte (byte v) { return v; }
+	public short PassAndReturnsShort (short v) { return v; }
+	public ushort PassAndReturnsUShort (ushort v) { return v; }
+	public int PassAndReturnsInt (int v) { return v; }
+	public uint PassAndReturnsUInt (uint v) { return v; }
+	public long PassAndReturnsLong (long v) { return v; }
+	public ulong PassAndReturnsULong (ulong v) { return v; }
+	public char PassAndReturnsChar (char v) { return v; }
+	public string PassAndReturnsString (string v) { return v; }
```
Comment (spouliot) [inline tests/managed/types.cs]: all of the above should be working, they just need to be added in the Xcode unit tests (to confirm and avoid regressions)

File: tests/managed/arrays.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#if NON_OBJC_SUPPORTED_TESTS
```
Comment (spouliot) [inline tests/managed/arrays.cs]: maybe `!TARGET_OBJC` ? or something similar.  We'll likely have some cases where other generator or platforms needs customization (temporary or permanent). @rolfbjarne ?

Comment (tritao) [inline tests/managed/arrays.cs]: Yeah I like that better..

File: tests/managed/types.cs (hunk 1 above)
Comment (tritao) [inline tests/managed/types.cs]: OK I can try adding them to Xcode tests.

File: tests/managed/arrays.cs (hunk 2 above)
Comment (rolfbjarne) [inline tests/managed/arrays.cs]: Yeah, I like `TARGET_OBJC` better too.

Comment (spouliot) [inline tests/managed/arrays.cs]: sold! :)

**Comment by tritao** [comment]
Not needed anymore since ObjC branch can now deal with everything besides arrays, I'll integrate those tests once array support is added.

**Pull request closed without merging**